builtin/credential/approle/path_role.go

```diff
@@ -1695,6 +1723,50 @@ func (b *backend) handleRoleSecretIDCommon(req *logical.Request, data *framework
 		return logical.ErrorResponse("bind_secret_id is not set on the role"), nil
 	}
 
+	cidrList := data.Get("cidr_list").(string)
+
+	// Validate the list of CIDR blocks
+	if cidrList != "" {
+		valid, err := cidrutil.ValidateCIDRListString(cidrList, ",")
+		if err != nil {
+			return nil, fmt.Errorf("failed to validate CIDR blocks: %q", err)
+		}
+		if !valid {
+			return logical.ErrorResponse("failed to validate CIDR blocks"), nil
+		}
+	}
+
+	// Parse the CIDR blocks into a slice
+	secretIDCIDRs := strutil.ParseDedupAndSortStrings(cidrList, ",")
+	if len(secretIDCIDRs) != 0 {
+		// If role has bound_cidr_list set, check if each CIDR block is
+		// a subset of at least one CIDR block registered under the
+		// role. If bound_cidr_list is not set on the role, then it
+		// means that the role allows any IP address, implying that the
+		// CIDR blocks on the secret ID are a subset of that of role's.
+		roleCIDRs := strutil.ParseDedupAndSortStrings(role.BoundCIDRList, ",")
+		if len(roleCIDRs) != 0 {
+			subset, err := cidrutil.SubsetBlocks(roleCIDRs, secretIDCIDRs)
+			if err != nil {
+				return nil, fmt.Errorf("failed to verify subset relationship between CIDR blocks on the role %q and CIDR blocks on the secret ID %q: %v", roleCIDRs, secretIDCIDRs, err)
+			}
+			if !subset {
+				return logical.ErrorResponse(fmt.Sprintf("CIDR blocks on the secret ID %q should be a subset of CIDR blocks on the role %q", secretIDCIDRs, roleCIDRs)), nil
+			}
+		}
+
+		if req.Connection == nil || req.Connection.RemoteAddr == "" {
```

Comment:
I'm not sure what this check is here for. My understanding is that at *login* time the IP should belong within the role and/or secret ID CIDRs, but this is checking at secret ID generation time.